great_expectations/cli/toolkit.py

```diff
@@ -158,25 +159,30 @@ def launch_jupyter_notebook(notebook_path: str) -> None:
 
 def get_validator(
     context: DataContext,
-    batch_request: Union[dict, BatchRequest],
+    batch_request: Union[dict, BatchRequest, RuntimeBatchRequest],
     suite: Union[str, ExpectationSuite],
 ) -> Validator:
     assert isinstance(
         suite, (str, ExpectationSuite)
     ), "Invalid suite type (must be ExpectationSuite) or a string."
 
-    if isinstance(batch_request, dict):
-        batch_request = BatchRequest(**batch_request)
-
     validator: Validator
     if isinstance(suite, str):
-        validator = context.get_validator(
-            batch_request=batch_request, expectation_suite_name=suite
-        )
+        if isinstance(batch_request, dict):
+            validator = context.get_validator(
```

Comment:
Not a huge issue but I kind of like instantiating a BatchRequest here if we encounter a dict and passing that through so that we can continue tightening up our internal interfaces (by accepting objects instead of a ton of params). I realize here `get_validator()` is on the context so may need those params regardless but just a thought.